src/mono/browser/runtime/loader/run.ts

```diff
@@ -375,16 +384,22 @@ export class HostBuilder implements DotnetHostBuilder {
 }
 
 export async function createApi (): Promise<RuntimeAPI> {
+    await createEmscripten(emscriptenModule);
+    return globalObjectsRoot.api;
+}
+
+let emscriptenPrepared = false;
+async function prepareEmscripten (moduleFactory: DotnetModuleConfig | ((api: RuntimeAPI) => DotnetModuleConfig)) {
+    if (emscriptenPrepared) {
+        return;
+    }
+    emscriptenPrepared = true;
     if (ENVIRONMENT_IS_WEB && loaderHelpers.config.forwardConsoleLogsToWS && typeof globalThis.WebSocket != "undefined") {
         setup_proxy_console("main", globalThis.console, globalThis.location.origin);
     }
     mono_assert(emscriptenModule, "Null moduleConfig");
     mono_assert(loaderHelpers.config, "Null moduleConfig.config");
-    await createEmscripten(emscriptenModule);
-    return globalObjectsRoot.api;
-}
 
-export async function createEmscripten (moduleFactory: DotnetModuleConfig | ((api: RuntimeAPI) => DotnetModuleConfig)): Promise<RuntimeAPI | EmscriptenModuleInternal> {
     // extract ModuleConfig
     if (typeof moduleFactory === "function") {
```

Comment:
Why do we need to run moduleFactory if we want to only download resources?

Reply:
It could change the location of the config file